wire-ios-request-strategy/Sources/Request Strategies/Conversation/Actions/SyncUsersActionHandlerTests.swift

```diff
@@ -0,0 +1,120 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+@testable import WireRequestStrategy
+
+final class SyncUsersActionHandlerTests: MessagingTestBase {
+
+    // MARK: - Properties
+
+    typealias RequestPayload = SyncUsersActionHandler.RequestPayload
+    typealias ResponsePayload = SyncUsersActionHandler.ResponsePayload
+
+    // MARK: - Request Generation
+
+    func testRequestGenerationForUnsupportedAPIVersions() {
+        // GIVEN
+        let sut = SyncUsersActionHandler(context: uiMOC)
+        let id = QualifiedID(uuid: .create(), domain: "example.com")
+        let action = SyncUsersAction(qualifiedIDs: [id])
+
+        // Test for each unsupported version
+        for version in [APIVersion.v0, .v1, .v2, .v3] {
+            // WHEN
+            let request = sut.request(for: action, apiVersion: version)
+
+            // THEN
+            XCTAssertNil(request, "Request should be nil for API version \(version)")
+        }
+    }
+
+    func testRequestGenerationForAPIVersionV4() {
+        // GIVEN
+        let sut = SyncUsersActionHandler(context: uiMOC)
+        let id = QualifiedID(uuid: .create(), domain: "example.com")
+        let action = SyncUsersAction(qualifiedIDs: [id])
+
+        // WHEN
+        let result = try! XCTUnwrap(sut.request(for: action, apiVersion: .v4))
+
+        // Then
+        XCTAssertNotNil(result, "Request should not be nil for API version .v4")
+        XCTAssertEqual(result.path, "/v4/list-users", "Incorrect path for API version .v4")
+        XCTAssertEqual(result.method, .post, "Incorrect HTTP method for API version .v4")
+        XCTAssertEqual(result.apiVersion, APIVersion.v4.rawValue, "Incorrect API version set in the request for .v4")
+
+        let payload = try! XCTUnwrap(RequestPayload(result))
+        XCTAssertEqual(payload, RequestPayload(qualified_users: [id]))
+    }
+
+    func testRequestGenerationForAPIVersionV5() {
+        // GIVEN
+        let sut = SyncUsersActionHandler(context: uiMOC)
+        let id = QualifiedID(uuid: .create(), domain: "example.com")
+        let action = SyncUsersAction(qualifiedIDs: [id])
+
+        // WHEN
+        let result = try! XCTUnwrap(sut.request(for: action, apiVersion: .v5))
+
+        // THEN
+        XCTAssertNotNil(result, "Request should not be nil for API version .v5")
+        XCTAssertEqual(result.path, "/v5/list-users", "Incorrect path for API version .v5")
+        XCTAssertEqual(result.method, .post, "Incorrect HTTP method for API version .v5")
+        XCTAssertEqual(result.apiVersion, APIVersion.v5.rawValue, "Incorrect API version set in the request for .v5")
+
+        let payload = try! XCTUnwrap(RequestPayload(result))
+        XCTAssertEqual(payload, RequestPayload(qualified_users: [id]))
+    }
+
+    // MARK: - Response Handling
```

Comment:
There should also be a test for the success.

Reply:
Done! @johnxnguyen